### ISSUE TITLE
Build issue with STM32L4 series

### ISSUE THREAD

**Issue by JeremySavonet** [UAVCAN/platform_specific_components#2]
Hi,

It seems that STM32L4 series have no APB1ENR register causing the build to fail.
Here is a dump of the issue:

```
3rdparty/libuavcan/libuavcan_drivers/stm32/driver/src/uc_stm32_clock.cpp: In function 'void uavcan_stm32::clock::init()':
3rdparty/libuavcan/libuavcan_drivers/stm32/driver/src/uc_stm32_clock.cpp:41:39: error: 'struct RCC_TypeDef' has no member named 'APB1ENR'; did you mean 'AHB1ENR'?
 #  define TIMX_RCC_ENR           RCC->APB1ENR
```

Regards,

**Comment by JeremySavonet** [comment]
This could be a patch if needed:

```
diff --git a/libuavcan_drivers/stm32/driver/src/uc_stm32_can.cpp b/libuavcan_drivers/stm32/driver/src/uc_stm32_can.cpp
index 2fe8ad67..ebd21a69 100644
--- a/libuavcan_drivers/stm32/driver/src/uc_stm32_can.cpp
+++ b/libuavcan_drivers/stm32/driver/src/uc_stm32_can.cpp
@@ -993,9 +993,9 @@ void CanDriver::initOnce()
         modifyreg32(STM32_RCC_APB1RSTR, RCC_APB1RSTR_CAN2RST, 0);
 # endif
 #else
-        RCC->APB1ENR  |=  RCC_APB1ENR_CAN1EN;
-        RCC->APB1RSTR |=  RCC_APB1RSTR_CAN1RST;
-        RCC->APB1RSTR &= ~RCC_APB1RSTR_CAN1RST;
+        RCC->APB1ENR1  |=  RCC_APB1ENR1_CAN1EN;
+        RCC->APB1RSTR1 |=  RCC_APB1RSTR1_CAN1RST;
+        RCC->APB1RSTR1 &= ~RCC_APB1RSTR1_CAN1RST;
 # if UAVCAN_STM32_NUM_IFACES > 1
         RCC->APB1ENR  |=  RCC_APB1ENR_CAN2EN;
         RCC->APB1RSTR |=  RCC_APB1RSTR_CAN2RST;
diff --git a/libuavcan_drivers/stm32/driver/src/uc_stm32_clock.cpp b/libuavcan_drivers/stm32/driver/src/uc_stm32_clock.cpp
index 1a4cd1e6..120343e6 100644
--- a/libuavcan_drivers/stm32/driver/src/uc_stm32_clock.cpp
+++ b/libuavcan_drivers/stm32/driver/src/uc_stm32_clock.cpp
@@ -38,10 +38,10 @@
 # endif

 # if UAVCAN_STM32_TIMER_NUMBER >= 2 && UAVCAN_STM32_TIMER_NUMBER <= 7
-#  define TIMX_RCC_ENR           RCC->APB1ENR
-#  define TIMX_RCC_RSTR          RCC->APB1RSTR
-#  define TIMX_RCC_ENR_MASK      UAVCAN_STM32_GLUE3(RCC_APB1ENR_TIM,  UAVCAN_STM32_TIMER_NUMBER, EN)
-#  define TIMX_RCC_RSTR_MASK     UAVCAN_STM32_GLUE3(RCC_APB1RSTR_TIM, UAVCAN_STM32_TIMER_NUMBER, RST)
+#  define TIMX_RCC_ENR           RCC->APB1ENR1
+#  define TIMX_RCC_RSTR          RCC->APB1RSTR1
+#  define TIMX_RCC_ENR_MASK      UAVCAN_STM32_GLUE3(RCC_APB1ENR1_TIM,  UAVCAN_STM32_TIMER_NUMBER, EN)
+#  define TIMX_RCC_RSTR_MASK     UAVCAN_STM32_GLUE3(RCC_APB1RSTR1_TIM, UAVCAN_STM32_TIMER_NUMBER, RST)
 # else
 #  error "This UAVCAN_STM32_TIMER_NUMBER is not supported yet"
 # endif
```

**Comment by pavel-kirienko** [comment]
Thanks, @JeremySavonet! The patch would break other MCUs though. Would it not be possible to implement the fix in a way that wouldn't have any destructive consequences for other MCUs? We would gladly accept that into the upstream.

**Comment by JeremySavonet** [comment]
Here is a cleaner fix without breaking build for other MCUs:
https://github.com/UAVCAN/libuavcan_stm32/pull/3

**Comment by pavel-kirienko** [comment]
Fixed in #3
